File: scripts/testSerial.py
import serial
import time
import sys

port = "/dev/tty.bluetooth-Bailey"
#port = "/dev/tty.bluetooth-Jamie"
#port = "/dev/tty.nxt8"
#port = "/dev/tty.Jamie-DevB"

ser = serial.Serial(port, 115200, timeout=None)
ser.flushInput()
ser.flushOutput()
#ser = serial.Serial(port, 115200)
package = "\x0c\x00\x80\x04\x00\x64\x07\x01\x00\x20\x00\x00\x00\x00"

#if len(sys.argv)>1 and sys.argv[1] == 'stop':
#  package = "\x0c\x00\x80\x04\x00\x00\x07\x01\x00\x20\x00\x00\x00\x00"
#  print 'stop'

x = ser.write(package)
print 'baudrate: ', ser.baudrate
print 'Write output: ', x
ser.flush()

time.sleep(2)

package = "\x0c\x00\x80\x04\x00\x00\x07\x01\x00\x20\x00\x00\x00\x00"
x = ser.write(package)
print 'Write output: ', x

#while True:
#    data = ser.read(9999)
#    if len(data) > 0:
#        print 'Got:', data
#
#    sleep(0.5)

ser.close()

File: app/static/js/test_serial.js
var port = "/dev/tty.bluetooth-Bailey";

var init = function() {
  chrome.experimental.serial.open(port, function(cInfo) {
    startReading(cInfo);
  });
}

var startTime;
var bytesRead;

var startReading = function(cInfo) {
  startTime=Date.now(); 
  read(cInfo);
}

var read = function(cInfo) {
  chrome.experimental.serial.read(cInfo.connectionId, function(data) {
    bytesRead+=data.bytesRead;
    var elapsed=Date.now-startTime;
    if (elapsed<20000) {
      read(cInfo);
    } else {
      console.log("read "+bytesRead+" in "+elapsed+" ms");
    }
  });
}

init();
